factcast-store-validation/src/test/java/org/factcast/store/pgsql/internal/listen/PgListenerTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © 2018 factcast (http://factcast.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.pgsql.internal.listen;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.function.Predicate;
+
+import org.factcast.store.pgsql.internal.PgConstants;
+import org.factcast.store.pgsql.internal.listen.PgListener.FactInsertionEvent;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.postgresql.PGNotification;
+import org.postgresql.core.Notification;
+import org.postgresql.jdbc.PgConnection;
+
+import com.google.common.eventbus.AsyncEventBus;
+
+@ExtendWith(MockitoExtension.class)
+public class PgListenerTest {
+
+    @Mock
+    PgConnectionSupplier ds;
+
+    @Mock
+    AsyncEventBus bus;
+
+    @Mock
+    PgConnection conn;
+
+    @Mock
+    PreparedStatement ps;
+
+    Predicate<Connection> tester = c -> true;
+
+    @Captor
+    ArgumentCaptor<PgListener> captor;
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void testCheckFails() throws SQLException {
+
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(ps);
+
+        tester = mock(Predicate.class);
+        Mockito.when(tester.test(any())).thenReturn(false, false, true, false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)